pkg/network/setup/netpod/netpod.go

```diff
@@ -149,13 +147,34 @@ func WithVMIIfaceStatuses(vmiIfaceStatuses []v1.VirtualMachineInstanceNetworkInt
 	}
 }
 
+func WithDynamicPodIfaceNamingEnabled(enabled bool) option {
+	return func(n *NetPod) {
+		n.dynamicPodIfaceNamingEnabled = enabled
+	}
+}
+
 func (n NetPod) Setup() error {
 	// Not all network bindings are processed in the network setup.
 	filteredNets, err := filterSupportedBindingNetworks(n.vmiSpecNets, n.vmiSpecIfaces)
 	if err != nil {
 		return err
 	}
 
+	if n.dynamicPodIfaceNamingEnabled {
+		ifaceStatusesByName := vmispec.IndexInterfaceStatusByName(n.vmiIfaceStatuses, func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool {
+			return ifaceStatus.Name != "" && ifaceStatus.PodInterfaceName != ""
+		})
+
+		filteredNets = vmispec.FilterNetworksSpec(filteredNets, func(net v1.Network) bool {
+			_, exists := ifaceStatusesByName[net.Name]
+			return exists
+		})
```

Comment:
Consider adding a scenario in which the pod interface name is missing for some networks.